dashboard/src/components/charts/StackedAreaChartTooltip.tsx

```diff
@@ -0,0 +1,153 @@
+'use client';
+
+import React from 'react';
+import { cn } from '@/lib/utils';
+import { getTrendInfo, formatDifference, defaultDateLabelFormatter } from '@/utils/chartUtils';
+import { type ComparisonMapping } from '@/types/charts';
+import { type GranularityRangeValues } from '@/utils/granularityRanges';
+
+interface PayloadEntry {
+  value: number;
+  dataKey: string;
+  name?: string;
+  color?: string;
+}
+
+interface StackedAreaChartTooltipProps {
+  active?: boolean;
+  payload?: any;
+  label?: string | number;
+  formatter?: (value: number) => string;
+  labelFormatter?: (date: string | number, granularity?: GranularityRangeValues) => string;
+  comparisonMap?: ComparisonMapping[];
+  granularity?: GranularityRangeValues;
+}
+
+export function StackedAreaChartTooltip({
+  active,
+  payload,
+  label,
+  formatter = (value) => value.toLocaleString(),
+  labelFormatter = defaultDateLabelFormatter,
+  comparisonMap,
+  granularity,
+}: StackedAreaChartTooltipProps) {
+  if (!active || !payload || !payload.length || !label) {
+    return null;
+  }
+
+  const hasComparison = !!comparisonMap;
+
+  const comparisonData = hasComparison
+    ? comparisonMap.find((mapping) => mapping.currentDate === Number(label))
+    : null;
+
+  const getCurrentValue = (entry: PayloadEntry): number => {
+    return entry.value || 0;
+  };
+
+  const getCompareValue = (entry: PayloadEntry): number => {
+    return comparisonData ? comparisonData.compareValues[entry.dataKey] || 0 : 0;
+  };
+
+  const currentTotal = payload.reduce((sum: number, entry: PayloadEntry) => sum + getCurrentValue(entry), 0);
+  const compareTotal = comparisonData
+    ? payload.reduce(
+        (sum: number, entry: PayloadEntry) => sum + (comparisonData.compareValues[entry.dataKey] || 0),
+        0,
+      )
+    : 0;
```

Comment:
This logic can be simplified with your helper functions:
```suggestion
  const currentTotal = payload.reduce((sum: number, entry: PayloadEntry) => sum + getCurrentValue(entry), 0);
  const compareTotal = payload.reduce(
    (sum: number, entry: PayloadEntry) => sum + getCompareValue(entry),
    0,
  );
```